fedimint-client/src/db.rs

```diff
@@ -262,3 +276,173 @@ impl_db_lookup!(
     key = ClientModuleRecovery,
     query_prefix = ClientModuleRecoveryPrefix
 );
+
+/// ClientMigrationMap is a BTreeMap that maps DatabaseVersions to async
+/// functions. These functions are expected to "migrate" the database from the
+/// keyed DatabaseVersion to DatabaseVersion + 1.
+pub type ClientMigrationMap = BTreeMap<
+    DatabaseVersion,
+    for<'r, 'tx> fn(
```

Comment:
`type MigrationFn = ...` or something it.

Reply:
Nice type with `for ...`, two lifetimes and pining and futures. Proper nice async Rust. :D

Reply:
Done